### PR TITLE
Address concerns from Guido Vranken fuzzers

### PR DESCRIPTION
**Pull request by keeshux**

### CODE REVIEW

File: TunnelKit/Sources/Core/RoutingTableEntry.m [hunk 1]
```diff
@@ -508,10 +521,12 @@ static inline uint32_t RoutingTableEntryAddress4(NSString *string)
         return("default");
     }
     
-    getnameinfo((struct sockaddr *)sa6, sa6->sin6_len, line, sizeof(line), NULL, 0, flag);
+    ASSERT_GETNAMEINFO(getnameinfo((struct sockaddr *)sa6, sa6->sin6_len, host, sizeof(host), NULL, 0, flag));
     
     if (masklen > 0) {
-        sprintf(line, "%s/%u", line, masklen);
+        sprintf(line, "%s/%u", host, masklen);
```
Comment (guidovranken) [inline TunnelKit/Sources/Core/RoutingTableEntry.m:527]: ASSERT_PRINTF

File: TunnelKit/Sources/Core/RoutingTableEntry.m [hunk 2]
```diff
@@ -508,10 +521,12 @@ static inline uint32_t RoutingTableEntryAddress4(NSString *string)
...
+    } else {
+        sprintf(line, "%s", host);
```
Comment (guidovranken) [inline TunnelKit/Sources/Core/RoutingTableEntry.m:529]: ASSERT_PRINTF

**Comment by keeshux** [comment]
@guidovranken good to merge?

File: TunnelKit/Sources/Core/RoutingTableEntry.m [hunk 3]
```diff
@@ -243,15 +248,20 @@ - (BOOL)matchesDestination:(NSString *)destination
...
     const int halfPrefix = (int)(self.prefix + 1);
     if (self.isIPv6) {
+        if (self.prefix == 128) {
+            NSLog(@"Can't partition single IPv6");
+            return @[self, self];
+        }
+        
         struct in6_addr saddr1, saddr2;
         char addr[INET6_ADDRSTRLEN];
         NSData *addressData = RoutingTableEntryAddress6(self.network);
```
Comment (guidovranken) [inline TunnelKit/Sources/Core/RoutingTableEntry.m:258]: NULL check of addressData?

```
if ( addressData == NULL ) return NULL;
```

Comment (guidovranken) [reply TunnelKit/Sources/Core/RoutingTableEntry.m:258]: Also NULL checks after calls to ```RoutingTableEntryAddress6``` in ```matchesDestination```. At least in my fuzzers I observe crashes if those checks are not there.

```RoutingTableEntryAddress4``` calls might need similar checks.

**Comment by guidovranken** [comment]
> @guidovranken good to merge?

After addressing my remarks about RoutingTableEntry.m, yes.

**Comment by keeshux** [comment]
Those `RoutingTable*` are a real drag.

**Pull request merged**